tests/test_openclhpp.cpp

```diff
@@ -4336,4 +4337,83 @@ void testTemplateGetImageRequirementsInfo()
 void testTemplateGetImageRequirementsInfo() {}
 #endif // cl_ext_image_requirements_info
 
+static cl_mem clCreateFromGLBuffer_testgetObjectInfo(cl_context context,
+                                                     cl_mem_flags flags,
+                                                     cl_GLuint bufobj,
+                                                     cl_int *errcode_ret,
+                                                     int num_calls)
+{
+    switch (num_calls) {
+    case 0:
+            TEST_ASSERT_EQUAL(0, bufobj);
+            break;
+    case 1:
+            TEST_ASSERT_EQUAL(1, bufobj);
+            break;
+    default:
+            break;
+    }
+
+    TEST_ASSERT_EQUAL_PTR(make_context(0), context);
+    TEST_ASSERT_EQUAL(0, flags);
+    if (errcode_ret)
+        *errcode_ret = CL_SUCCESS;
+    return make_mem(0);
+}
+
+static cl_int clGetGLObjectInfo_testgetObjectInfo(cl_mem memobj,
+                                                  cl_gl_object_type *type,
+                                                  cl_GLuint *gl_object_name,
+                                                  int num)
+{
+    cl_int ret = CL_SUCCESS;
+    TEST_ASSERT_EQUAL(memobj, make_mem(0));
+    switch (num) {
+    case 0:
+        TEST_ASSERT_EQUAL(CL_GL_OBJECT_TEXTURE2D_ARRAY, *type);
+        *type = CL_GL_OBJECT_BUFFER;
+        break;
+    case 1:
+        TEST_ASSERT_EQUAL(CL_GL_OBJECT_BUFFER, *type);
+        *type = CL_GL_OBJECT_TEXTURE2D_ARRAY;
+        ret = CL_DEVICE_NOT_FOUND;
+        break;
+    default:
+        break;
+    }
+
+    *gl_object_name = 0;
+    return ret;
+}
+
+cl_int clReleaseMemObject_testgetObjectInfo(cl_mem memobj, int ret)
+{
+    return CL_SUCCESS;
+}
+
+void testgetObjectInfo() {
+    cl_mem_flags flags = 0;
+    cl_int err = 0;
+    cl_GLuint bufobj = 0;
+    cl_mem memobj = make_mem(0);
+    cl_gl_object_type type = CL_GL_OBJECT_TEXTURE2D_ARRAY;
+    clGetGLObjectInfo_StubWithCallback(clGetGLObjectInfo_testgetObjectInfo);
+    clCreateFromGLBuffer_StubWithCallback(
+        clCreateFromGLBuffer_testgetObjectInfo);
+    clReleaseMemObject_StubWithCallback(clReleaseMemObject_testgetObjectInfo);
```

Comment:
If you use this then you won't need to define a specific stub for `clReleaseMemObject`:

```suggestion
    clReleaseMemObject_ExpectAndReturn(make_mem(0), CL_SUCCESS);
```